File: app/src/main/java/com/kota/ASFramework/PageController/ASAnimationRunner.java
package com.kota.ASFramework.PageController;

import android.view.View;
import android.view.animation.Animation;
import com.kota.ASFramework.Thread.ASRunner;

/* loaded from: classes.dex */
public abstract class ASAnimationRunner {
  Animation _animation;

  abstract View getTargetView();

  abstract void onAnimationStartFail();

  public ASAnimationRunner(Animation animation) {
    this._animation = null;
    this._animation = animation;
  }

  public void start() {
    ASRunner.runInNewThread(()->{
        int i = 0;
        boolean success = false;
        while (true) {
          if (i >= 10) {
            break;
          }
          View target_view = ASAnimationRunner.this.getTargetView();
          if (target_view != null) {
            ASAnimationRunner.this.animate();
            success = true;
            break;
          }
          i++;
          try {
            Thread.sleep(10L);
          } catch (InterruptedException e) {
            e.printStackTrace();
          }
        }
        if (!success) {
          ASAnimationRunner.this.fail();
        }
    });
  }

  /* JADX INFO: Access modifiers changed from: private */
  public void fail() {
    new ASRunner() { // from class: com.kumi.ASFramework.PageController.ASAnimationRunner.2
      @Override // com.kumi.ASFramework.Thread.ASRunner
      public void run() {
        ASAnimationRunner.this.onAnimationStartFail();
      }
    }.runInMainThread();
  }

  /* JADX INFO: Access modifiers changed from: private */
  public void animate() {
    new ASRunner() { // from class: com.kumi.ASFramework.PageController.ASAnimationRunner.3
      @Override // com.kumi.ASFramework.Thread.ASRunner
      public void run() {
        View target_view = ASAnimationRunner.this.getTargetView();
        if (target_view != null) {
          target_view.startAnimation(ASAnimationRunner.this._animation);
        }
      }
    }.runInMainThread();
  }
}

File: app/src/main/java/com/kota/ASFramework/PageController/ASAnimationRunner.kt
package com.kota.ASFramework.PageController

import android.view.View
import android.view.animation.Animation
import com.kota.ASFramework.Thread.ASRunner
import com.kota.ASFramework.Thread.ASRunner.Companion.runInNewThread

/* loaded from: classes.dex */
abstract class ASAnimationRunner(animation: Animation?) {
    var _animation: Animation? = null

    abstract val targetView: View?

    abstract fun onAnimationStartFail()

    init {
        this._animation = animation
    }

    fun start() {
        runInNewThread(Runnable {
            var i = 0
            var success = false
            while (true) {
                if (i >= 10) {
                    break
                }
                val target_view = this@ASAnimationRunner.targetView
                if (target_view != null) {
                    this@ASAnimationRunner.animate()
                    success = true
                    break
                }
                i++
                try {
                    Thread.sleep(10L)
                } catch (e: InterruptedException) {
                    e.printStackTrace()
                }
            }
            if (!success) {
                this@ASAnimationRunner.fail()
            }
        })
    }

    private fun fail() {
        object : ASRunner() {
            // from class: com.kota.ASFramework.PageController.ASAnimationRunner.2
            // com.kota.ASFramework.Thread.ASRunner
            public override fun run() {
                this@ASAnimationRunner.onAnimationStartFail()
            }
        }.runInMainThread()
    }

    private fun animate() {
        object : ASRunner() {
            // from class: com.kota.ASFramework.PageController.ASAnimationRunner.3
            // com.kota.ASFramework.Thread.ASRunner
            public override fun run() {
                val target_view = this@ASAnimationRunner.targetView
                if (target_view != null) {
                    target_view.startAnimation(this@ASAnimationRunner._animation)
                }
            }
        }.runInMainThread()
    }
}